pennylane_lightning/core/bindings/BindingsUtils_nb.hpp

```diff
@@ -25,7 +25,7 @@ namespace Pennylane::NanoBindings::Utils {
  */
 template <typename VectorT>
 nb::ndarray<VectorT, nb::numpy, nb::c_contig>
-createNumpyArrayFromVector(const std::vector<VectorT> &data,
+createNumpyArrayFromVector(std::vector<VectorT> &&data,
```

Comment:
Nice! Is there any concern w/ passing a tmp reference here?

Reply:
I will defer to @jzaia18 on that.